src/main/java/com/minecolonies/coremod/blocks/BlockBarracksTowerSubstitution.java

```diff
@@ -64,18 +64,6 @@ public BlockState rotate(@NotNull final BlockState state, final Rotation rot)
         return state.setValue(FACING, rot.rotate(state.getValue(FACING)));
     }
 
-    /**
-     * @deprecated (Remove this as soon as minecraft offers anything better).
-     */
-    @SuppressWarnings(DEPRECATION)
-    @NotNull
-    @Override
-    @Deprecated
-    public BlockState mirror(@NotNull final BlockState state, final Mirror mirrorIn)
-    {
-        return state.rotate(mirrorIn.getRotation(state.getValue(FACING)));
-    }
```

Comment:
Possibly of note is that most other MineColonies blocks include similar code, either directly or via inheritance from `HorizontalBlock`...

Reply:
public abstract class AbstractBlockHut<B extends AbstractBlockHut<B>> extends AbstractBlockMinecolonies<B> implements IBuilderUndestroyable, IAnchorBlock

Reply:
Yes, AbstractBlockHut doesn't.  I was referring to things like barrels, racks, graves, flags, timber frames, etc.  (And sort of deco controllers, but they override `mirror` differently anyway so don't really count.)

Shingles also seem to get messed up (as noted in discord), although those don't have a custom mirror, they get it from vanilla stairs.  So it seems like there's something more systemic going wrong.

Reply:
This here is for the schematic misalignment, not of blocks weirdly running into issues when mirrored.

Reply:
Ok, but I just tested on this branch and the mirroring is still wrong.  Asian barracks, level 5, no rotation.  With this code change it fixes the incorrect mirroring of the front two towers but not that of the back two towers.  Again, it mostly just looks like structurize is mirroring on the wrong axis.

Reply:
Okay I checked the code and I know what's up. The barrackstowerplaceholders do not store the mirror anywhere. All other buildings store the mirror in the tileEntity and use the rotation of the block. So this code doesn't actually help because the barrackstowersubs don't have a tileEntity. So, the real fix is removing the barrackstowesubstitution and use the new integrated system for this properly.

Reply:
That may also be true, but it's still getting the block facing incorrect.  With this change reverted, it now changes the facing of the front two and still doesn't change the facing of the back two, which is worse.

Reply:
Getting someone to rescan the barracks with barrackstower instead of placeholder to fix this for good.